internal/component/mimir/alerts/kubernetes/alerts.go

```diff
@@ -298,6 +303,19 @@ func (c *Component) init() error {
 		return err
 	}
 
+	c.matcherStrategy = monitoringv1.AlertmanagerConfigMatcherStrategyType(c.args.AlertmanagerConfigMatcherStrategy)
+	if c.matcherStrategy != monitoringv1.OnNamespaceConfigMatcherStrategyType &&
+		c.matcherStrategy != monitoringv1.OnNamespaceExceptForAlertmanagerNamespaceConfigMatcherStrategyType &&
+		c.matcherStrategy != monitoringv1.NoneConfigMatcherStrategyType {
+		return fmt.Errorf("invalid alertmanagerconfig_matcher_strategy: %s", c.args.AlertmanagerConfigMatcherStrategy)
+	}
```

Comment:
This would look better with a switch statement.

Reply:
I pushed a change, is that what you mean?